api/profile/profile.go

```diff
@@ -71,9 +71,6 @@ type Profile struct {
 	// Username is the Teleport username for the client.
 	Username string `yaml:"user,omitempty"`
 
-	// AuthType (like "google")
-	AuthType string `yaml:"auth_type,omitempty"`
```

Comment:
The paranoid dev in me worries that this might break someones' profile unmarshal out there, but after a quick lookup it really does look like this was never used.

Reply:
We are not using strict flag when unmarshaling so it is safe. 
I have even added UT to verify it `TestSafeRemovalOfAuthType` however I am planning to remove that UT when PR will be approved (i think it does not make any sense to keep it later).

Reply:
Thanks for double-checking. Feel free to either delete or keep it around to avoid regressions, your call.